include/maliput/math/kd_tree.h

```diff
@@ -130,45 +230,48 @@ struct NodeCmp {
 ///
 /// Inspired on https://rosettacode.org/wiki/K-d_tree.
 ///
+/// @tparam CRTP Derived class. It follows the curiously recurring template pattern.
 /// @tparam Coordinate Data type being used, must have:
 /// - operator[] for accessing the value in each dimension.
 /// @tparam Dimension Dimension of the KD-tree.
+/// @tparam Region The type of the region.
 /// @tparam Distance A functor used for getting the distance between two coordinates. By default,
 /// details::SquaredDistance is used.
 /// @tparam NodeCmp A functor used for comparing two nodes at certain index/dimension. By default, details::NodeCmp is
 /// used.
-template <typename Coordinate, std::size_t Dimension,
+template <typename CRTP, typename Coordinate, std::size_t Dimension, typename Region = BoundingRegion<Coordinate>,
           typename Distance = details::SquaredDistance<Coordinate, Dimension>,
           typename NodeCmp = details::NodeCmp<Dimension>>
-class KDTree {
+class KDTreeBase {
```

Comment:
Shall we move this class to the detail namespace?

Reply:
Done.